### PR TITLE
[#911] Add validation to migration throttling inputs and add CPU throttling field

### PR DESCRIPTION
**Pull request by mzazrivec**
Initial screen
migration01

Validation error
migration02

Validation pass
migration03

Fixes: https://github.com/ManageIQ/manageiq-v2v/issues/911

Associated RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1690851

### CODE REVIEW

File: app/javascript/react/screens/App/Settings/__tests__/__snapshots__/SettingsReducer.test.js.snap [hunk 1]
```diff
@@ -287,8 +287,10 @@ Object {
   "isSavingSettings": false,
   "postConversionHostsResults": Array [],
   "savedSettings": Object {
+    "cpu_limit_per_host": undefined,
```
Comment (mturley) [inline app/javascript/react/screens/App/Settings/__tests__/__snapshots__/SettingsReducer.test.js.snap]: We should add these fields to the SettingsReducer tests so they appear in the snapshots here.

File: app/javascript/react/screens/App/common/forms/TextInputWithCheckbox.js [hunk 2]
```diff
@@ -37,18 +43,11 @@ class TextInputWithCheckbox extends React.Component {
...
-              id={id}
-              defaultValue={value}
-              readOnly={!this.state.inputEnabled}
-              onChange={event => onChange(event.target.value)}
-            />
-            <div className="input-group-addon postfix-label">{postfix}</div>
-          </div>
+          <InputGroup>
+            <FormControl type="text" name={id} id={id} readOnly={!this.state.inputEnabled} {...input} />
+            <InputGroup.Addon>{postfix}</InputGroup.Addon>
```
Comment (mturley) [inline app/javascript/react/screens/App/common/forms/TextInputWithCheckbox.js]: I haven't had a chance yet to test this locally, but based on your screenshots it doesn't look like this postfix is rendering anymore?

**Comment by mzazrivec** [comment]
> What happened to those postfix labels, though?

Those postfix labels do work, I just forgot to update the screenshots with the working version. Fixed now.

File: app/javascript/react/screens/App/common/forms/TextInputWithCheckbox.js (hunk 2 above)
Comment (mzazrivec) [reply app/javascript/react/screens/App/common/forms/TextInputWithCheckbox.js]: Yup, those labels are being rendered correctly (I forgot to update screenshots in the initial comment).

File: app/javascript/react/screens/App/Settings/__tests__/__snapshots__/SettingsReducer.test.js.snap (hunk 1 above)
Comment (mzazrivec) [reply app/javascript/react/screens/App/Settings/__tests__/__snapshots__/SettingsReducer.test.js.snap]: You're right. Fixed.

**Comment by mturley** [comment]
This PR is associated with this BZ targeted for 5.10.4: https://bugzilla.redhat.com/show_bug.cgi?id=1690851

**Comment by mturley** [comment]
@mzazrivec sorry for the additional churn, but based on the call with @fdupont-redhat earlier today, it looks like CPU throttling will make it in for 5.10.4 but the network throttling will need to wait until 5.10.5.

So... deja vu... can we comment out the network field and turn this PR into CPU throttling only, and then open another PR to re-enable the network field?

**Comment by mzazrivec** [comment]
@mturley Sure, I removed the network throttling in a separate commit, which can be later reverted (once backend support is in).

**Comment by mturley** [comment]
Perfect, thanks @mzazrivec. I moved the network part of the issue to #940 so we can still close #911 here.

**Comment by mturley** [comment]
@mzazrivec, last thing, I promise.... I was giving your changes a final review and I noticed we never right-aligned the field names for the Concurrent Migrations section like they are in [the mockups](https://docs.google.com/presentation/d/15r00yDsnGo0K0owHxUb842AA-PnCfO6u5bc4NZNCgeU/edit#slide=id.g51ee9915c7_0_1). We should have those two field labels right-aligned, and the ones with the checkboxes left-aligned. If you could make that change I'll approve the PR, sorry I didn't notice sooner.

**Comment by mzazrivec** [comment]
@mturley Initially, I did try to implement it that way, but it did not look very good in my opinion:

form02

vs. all labeles left aligned:
form01

I can make the change, if you insist, but I'd rather keep it the way it looks right now, since mixing the label alignments produces a bit funny look & fell (in my opinion of course).

**Comment by mturley** [comment]
I think I agree with you, but I'll let @vconzola make the final decision.

**Comment by vconzola** [comment]
Per our discussion in the v2v standup meeting, @mturley will try to make a simple change to the CSS class to use top aligned labels. But if it's too much, let's just leave them left aligned.

**Comment by mturley** [comment]
@mzazrivec here's a commit that switches the form to a vertical layout: https://github.com/mturley/manageiq-v2v/commit/5ba960c05c931c48f4a20bc61a5130eee062e4a9

It looks like this. @vconzola does this look good to you?

Screenshot 2019-04-18 11 53 58

**Comment by vconzola** [comment]
Other than my few comments in Slack, LGTM. Ship it.

**Comment by mturley** [comment]
Ok @mzazrivec I made a couple other tweaks to the commit based on feedback from Vince, here's the final version to cherry-pick: https://github.com/mturley/manageiq-v2v/commit/acfd88f9152a79b28eb853417a20832195830ba9

**Comment by mturley** [comment]
The backend PR this depended on is also now merged (https://github.com/ManageIQ/manageiq/pull/18576).

**Comment by mturley** [comment]
I'm going to go ahead and merge this one as-is so it can make the GA build, I'll open a separate PR to change the layout to vertical.

**Comment by simaishi** [comment]
Hammer backport details:

```
$ git log -1
commit ea0aa2770cedba6d7c4098891e79d64244945652
Author: Mike Turley <mike.turley@alum.cs.umass.edu>
Date:   Thu Apr 18 14:56:43 2019 -0400

    Merge pull request #915 from mzazrivec/add_validation_to_migration_throttling_inputs
    
    [#911] Add validation to migration throttling inputs and add CPU throttling field
    
    (cherry picked from commit 03106c9ca7a9c72d54e70c1eddfac9af53f3404f)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1702085
```

**Comment by simaishi** [comment]
Reverted the backport.

**Pull request merged**